scripts/lib-franklin.js

```diff
@@ -127,25 +127,46 @@ export function toCamelCase(name) {
  * Replace icons with inline SVG and prefix with codeBasePath.
  * @param {Element} element
  */
-export function decorateIcons(element = document) {
-  element.querySelectorAll('span.icon').forEach(async (span) => {
-    if (span.classList.length < 2 || !span.classList[1].startsWith('icon-')) {
-      return;
-    }
-    const icon = span.classList[1].substring(5);
-    // eslint-disable-next-line no-use-before-define
-    const resp = await fetch(`${window.hlx.codeBasePath}/icons/${icon}.svg`);
-    if (resp.ok) {
-      const iconHTML = await resp.text();
-      if (iconHTML.match(/<style/i)) {
-        const img = document.createElement('img');
-        img.src = `data:image/svg+xml,${encodeURIComponent(iconHTML)}`;
-        span.appendChild(img);
-      } else {
-        span.innerHTML = iconHTML;
+export async function decorateIcons(element) {
+  const symbols = {};
+  let svgSprite = document.getElementById('franklin-svg-sprite');
+  if (!svgSprite) {
+    const div = document.createElement('div');
+    div.innerHTML = '<svg xmlns="http://www.w3.org/2000/svg" id="franklin-svg-sprite" style="display: none"></svg>';
+    svgSprite = div.firstElementChild;
+    document.body.append(div.firstElementChild);
+  }
+
+  const icons = [...element.querySelectorAll('span.icon')];
+
+  return Promise.all(icons.map(async (span) => {
+    const iconName = span.classList[1].substring(5);
+    let html;
+    if (!symbols[iconName]) {
+      symbols[iconName] = true;
```

Comment:
I tried using this code in https://github.com/hlxsites/maidenform/issues/114, but I think your latest commit broke the loading here.

If `symbols[iconName]` is `true`, `html` will be undefined.

In my code, I addressed this by saving the `html` variable to the map `symbols[iconName]`, instead of storing a boolean, and setting the parent `innerHTML` to this.